react/features/connection-indicator/middleware.web.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+
+import { downloadJSON } from '../../../modules/util/helpers';
+import getRoomName from '../base/config/getRoomName';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { CONNECTION_INDICATOR_SAVE_LOGS } from './actionTypes';
+
+/**
+ * Implements the middleware of the feature connection-indicator.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+// eslint-disable-next-line no-unused-vars
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case CONNECTION_INDICATOR_SAVE_LOGS: {
+        // this can be called from console and will not have reference to this
+        // that's why we reference the global var
+        const logs = store.getState()['features/base/connection'].connection.getLogs();
```

Comment:
Since you don't use this action to change the store maybe you can move the code from this middleware directly into the action itself and convert the action to async. Here's an example of async action - https://github.com/jitsi/jitsi-meet/blob/45570bc0e7d1722f7cce2a497e14d4baf0c0cec3/react/features/base/devices/actions.js#L70 .

This way you won't need middleware at all and actionType too. I guess this will simplify the  code. WDYT?